src/spectral/rulesets/custom-rules/content-entry-contains-schema.yaml

```diff
@@ -0,0 +1,12 @@
+rules:
```

Comment:
In hindsight, I think it may not be necessary to break the rules into different files. In different conversations about adding custom Sepctral rules, people mentioned that we may not want to clutter the `.defaultsForSpectral.yaml` file with custom rules.

I would say breaking custom rules in separate files has the following benefits:

1. Keeps the `.defaultsForSpectral.yaml` small and digestable.
2. Test files related to each custom rule are easier to create/find/update because test files map to a corresponding file with the same name.
3. Small benefit: When we add custom functions, some rules will need to specify the custom function it uses in a `functions` field. If the custom rules are in separate files, each real will clearly define the list of custom functions it uses rather than have a global list of custom functions in `.defaultsForSpectral.yaml`. This is only a small benefit because the custom rule will explicitly say which custom function it uses in the rule definition.

Do you think the reasons above are enough to introduce a more complicated structure now, or should we consider breaking rules into separate files as it becomes apparent that it's necessary?

Reply:
My take:
1. This just trades complexity within one file for the complexity of having multiple files -- which includes the extra logic in this PR for merging the multiple files.
2. Test files can be named for the rule -- I think this provides equal correlation between the rules and test files.
3. Valid but I'm not sure sufficient rationale for separate files.

I'll add to this that Spectral follows the "all rules in a single file" structure:

https://github.com/stoplightio/spectral/blob/develop/src/rulesets/oas/index.json

So I still think the single file approach is superior.

Reply:
Yes, I think you're right, and I will make the updates to move forward with a single file approach.